portmapper/mapper.go

```diff
@@ -170,11 +170,13 @@ func (pm *PortMapper) MapRange(container net.Addr, hostIP net.IP, hostPortStart,
 		return nil
 	}
 
-	if err := m.userlandProxy.Start(); err != nil {
-		if err := cleanup(); err != nil {
-			return nil, fmt.Errorf("Error during port allocation cleanup: %v", err)
+	if hostIP.To4() != nil {
```

Comment:
why are we skipping this section for ipv6

Reply:
This function is called twice for each port which cases a problem with the userland proxy which tries to bind the port twice.
Currently this is handled by only start the user land proxy on the IPv4 port forwarding.
At the moment I have no idea how to handle this in a proper way.

Reply:
are you testing this with the user land proxy enabled ?
does this work with it disabled ?
we can add a TODO for the user land proxy rather than add the `hostIP.To4()` checks in the code

Reply:
I simply run the unit test which fail, so I thing the user land proxy is enabled there.

In part of the rework of the allocatePorts I found maybe a better solution.
Now the user land proxy will only be enabled for the IPv4 allocation and not for IPv6 (see comment below)